docs/drafts/controlling-catalog-selection.md

```diff
@@ -0,0 +1,238 @@
+3. **Deprecated bundles are deprioritized.** If non-deprecated bundles are available, deprecated ones are ignored.
+
+### Handling Ambiguity Errors
+
+If the system cannot resolve to a single bundle due to ambiguity, it will generate an error. You can resolve this by:
+
+- **Refining your catalog selection criteria.**
+- **Adjusting catalog priorities.**
+- **Ensuring that only one bundle matches your package name and version requirements.**
+
+## End to End Example
+
+1. **Create or Update `ClusterCatalogs` with Appropriate Labels and Priority**
+
+   ```yaml
+   apiVersion: catalogd.operatorframework.io/v1alpha1
+   kind: ClusterCatalog
+   metadata:
+     name: catalog-a
+     labels:
+       redhat.com/support: "true"
+   spec:
+     priority: 1000
+     source:
+       type: image
+       image:
+         ref: quay.io/example/catalog-a:latest
+   ```
+
+   ```yaml
+   apiVersion: catalogd.operatorframework.io/v1alpha1
+   kind: ClusterCatalog
+   metadata:
+     name: catalog-b
+     labels:
+       redhat.com/support: "false"
+   spec:
+     priority: 500
+     source:
+       type: image
+       image:
+         ref: quay.io/example/catalog-b:latest
+   ```
+   NB: an `olm.operatorframework.io/metadata.name` label will be added automatically to ClusterCatalogs when applied
+
+
+2. **Create a `ClusterExtension` with Catalog Selection**
+
+   ```yaml
+   apiVersion: olm.operatorframework.io/v1alpha1
+   kind: ClusterExtension
+   metadata:
+     name: install-my-operator
+   spec:
+     packageName: my-operator
+     catalog:
+       selector:
+         matchLabels:
+           redhat.com/support: "true"
+   ```
+
+3. **Apply the Resources**
+
+   ```shell
+   kubectl apply -f catalog-a.yaml
+   kubectl apply -f catalog-b.yaml
+   kubectl apply -f install-my-operator.yaml
+   ```
+
+4. **Verify the Installation**
+
+   Check the status of the `ClusterExtension`:
+
+   ```shell
+   kubectl get clusterextension install-my-operator -o yaml
+   ```
+
+   The status should indicate that the bundle was resolved from `catalog-a` due to the higher priority and matching label.
+
+## Important Notes
+
+- **Default Behavior**: If you do not specify any catalog selection criteria, the system may select any available catalog that provides the requested package, and the choice is undefined.
+- **Logical AND of Selectors**: When using both `matchLabels` and `matchExpressions`, catalogs must satisfy all criteria.
+- **Deprecation Status**: Non-deprecated bundles are preferred over deprecated ones during resolution.
+- **Error Messages**: The system will update the `.status.conditions` of the `ClusterExtension` with meaningful messages if resolution fails due to ambiguity or no catalogs being selected.
+
+## References
+
+- [Minimal Controls for Selecting Catalogs to Resolve From](https://github.com/operator-framework/operator-controller/issues/1028)
```

Comment:
Add another reference to upstream kubernetes labels and selectors doc? https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

Reply:
done in e6864f8a